### PR TITLE
// Security CSRF tokens sf #BOOM-206

### PR DESCRIPTION
**Pull request by xGouley**

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
@xGouley we already have a csrf protection provided by Symfony framework, why do you need a specific one ?

File: src/PrestaShopBundle/Security/Admin/TokenizedController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Interface TokenizedController: A safe controller should allow POST only is a CSRF token is present.
+ *
+ * This interface will trigger a pre-action listener to check if:
+ * - the HTTP query is a POST method.
+ * - then if the right token is present.
+ *
+ * @package PrestaShopBundle\Security\Admin
+ */
+interface TokenizedController
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Security/Admin/TokenizedController.php]: Should be `TokenizedControllerInterface` regarding the Symfony [coding standards](http://symfony.com/doc/current/contributing/code/standards.html).

File: src/PrestaShopBundle/Security/Admin/TokenizedControllerListener.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ * @package PrestaShopBundle\Security\Admin
+ */
+class TokenizedControllerListener
+{
+    /**
+     * @var CsrfTokenManager
+     */
+    private $csrfProvider;
+
+    public function __construct(CsrfTokenManager $csrfProvider)
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Security/Admin/TokenizedControllerListener.php]: be careful, $csrfProvider class do exists, and doesnt have the same use.

Let's call it $csrfTokenManager :)

**Comment by xGouley** [comment]
"we already have a csrf protection provided by Symfony framework, why do you need a specific one ?" --> it's not a specific one, it's the Symfony's one.

To have an answer abotu WHY, please contact the security leader that ask for it.

File: src/PrestaShopBundle/Security/Admin/TokenizedController.php (hunk 1 above)
Comment (xGouley) [inline src/PrestaShopBundle/Security/Admin/TokenizedController.php]: Traits has been banned by previous decisions on this architecture. Cannot use them.

File: src/PrestaShopBundle/Security/Admin/TokenizedControllerListener.php (hunk 2 above)
Comment (xGouley) [inline src/PrestaShopBundle/Security/Admin/TokenizedControllerListener.php]: :+1: Noticed, I will fix this

**Comment by Quetzacoalt91** [comment]
Tokenized URLs have been implemented with the PR https://github.com/PrestaShop/PrestaShop/pull/5912
This one can be closed now.

Regards

**Pull request closed without merging**